### ISSUE TITLE
Objects that pass the spread syntax can no longer be assigned to assignable types.

### ISSUE THREAD

**Issue by sounisi5011** [microsoft/TypeScript#49235]
# Bug Report

### 🔎 Search Terms

+ PropertyDescriptorMap
+ PropertyDescriptor
+ Spread

### 🕗 Version & Regression Information

- This has changed since version 4.6

### ⏯ Playground Link

Playground link with relevant code:

+ [Nightly (fail)](https://www.typescriptlang.org/play?ts=next#code/CYUwxgNghgTiAEYD2A7AzgF3gBxk7IMGAliGgIwBc8A3vANoAe1mMxKA5gLrUAKeBIgE8AImTBtsGJDHgBfANwAoJQHkARgCtwGAHSgAZuxD98hEmQAUNOQBocA86QoBKZUuTosdXb9xmiZwAmeXgAXgcAiwplDW0wPUNjU0Fo6ztI1OC3JSA)
+ [4.7.0-beta (fail)](https://www.typescriptlang.org/play?ts=4.7.0-beta#code/CYUwxgNghgTiAEYD2A7AzgF3gBxk7IMGAliGgIwBc8A3vANoAe1mMxKA5gLrUAKeBIgE8AImTBtsGJDHgBfANwAoJQHkARgCtwGAHSgAZuxD98hEmQAUNOQBocA86QoBKZUuTosdXb9xmiZwAmeXgAXgcAiwplDW0wPUNjU0Fo6ztI1OC3JSA)
+ [4.6.4 (fail)](https://www.typescriptlang.org/play?ts=4.6.4#code/CYUwxgNghgTiAEYD2A7AzgF3gBxk7IMGAliGgIwBc8A3vANoAe1mMxKA5gLrUAKeBIgE8AImTBtsGJDHgBfANwAoJQHkARgCtwGAHSgAZuxD98hEmQAUNOQBocA86QoBKZUuTosdXb9xmiZwAmeXgAXgcAiwplDW0wPUNjU0Fo6ztI1OC3JSA)
+ [4.5.5 (passed)](https://www.typescriptlang.org/play?ts=4.5.5#code/CYUwxgNghgTiAEYD2A7AzgF3gBxk7IMGAliGgIwBc8A3vANoAe1mMxKA5gLrUAKeBIgE8AImTBtsGJDHgBfANwAoJQHkARgCtwGAHSgAZuxD98hEmQAUNOQBocA86QoBKZUuTosdXb9xmiZwAmeXgAXgcAiwplDW0wPUNjU0Fo6ztI1OC3JSA)

### 💻 Code

```ts
declare const properties1: { [x: string]: PropertyDescriptor };

// This will pass. This is because the type '{ [x: string]: PropertyDescriptor }' can be assigned to the type 'PropertyDescriptorMap'.
Object.defineProperties({}, properties1);

// But when it passes the spread syntax, it fails even though the types are the same.
const { ...properties2 } = properties1;
Object.defineProperties({}, properties2);
```

### 🙁 Actual behavior

I get the following error:

```
Argument of type '{ [x: string]: PropertyDescriptor; }' is not assignable to parameter of type 'PropertyDescriptorMap & ThisType<any>'.
  Type '{ [x: string]: PropertyDescriptor; }' is not assignable to type 'PropertyDescriptorMap'.
    Index signature for type 'symbol' is missing in type '{ [x: string]: PropertyDescriptor; }'.
```

But, type `{ [x: string]: PropertyDescriptor; }` is assignable to type `PropertyDescriptorMap`. Objects copied with spread syntax cannot be assigned even though they have the same type.

### 🙂 Expected behavior

This type error does not occur. For the same type, it should be consistently assignable / non-assignable anywhere.

**Comment by MartinJohns** [comment]
I don't know if this is intentional, but a quick workaround is to write `{ [x: string | symbol]: PropertyDescriptor }`.

**Comment by sounisi5011** [comment]
Yes. But in the actual code, we may use the spread syntax when we want to exclude some properties.

https://github.com/sounisi5011/npm-packages/blob/fdf42d3f512f5ed3cb24ee7c5056f96eeafbd014/packages/check-pid-file/src/index.ts#L15

The same error occurs in this case as well. Such code would require a type hack like the one below to set the proper type:

```ts
const oldErrorProperties = Object.getOwnPropertyDescriptors(oldError);
const { message: _1, stack: _2, ...newErrorProperties } =
    oldErrorProperties as (
        & typeof oldErrorProperties
        & { [x: symbol]: PropertyDescriptor }
    );
Object.defineProperties(newError, newErrorProperties);
```
[Playground](https://www.typescriptlang.org/play?ts=4.6.4#code/CYUwxgNghgTiAEYD2A7AzgF3kiwCiMMSMAXPAUTPAGTwDe8mUYA1mZjAJYoDm8AvgG4AUMOTosOfIWIAFIgAcQMDJxBp4AXngB5AEYArcBgB0PEBh0B3FPKRKVATwAi6sFwUZiaABRSKxACUIuKY9PAAtupoUOZkAPoAjAA0jBjMbPDxAEypJvn+MjB2DqrqOhGcGBggwIkCWti4AcWKymVoIvpGYKagAGbcICXtar50-KmFlCMqYxVVNXXBoqFYDFFoMXFZAMypTKwJACx5Bc1Fsx0L1bXZDZrC8M9N0jNtc+rwUBo+Ty8A2gYRxKJD9V4tK5jf4A560BgAbQAHuxHBE9DgALpkK4uNweLxUfgw+ArbrGEwDIZQ9Q+CZTC7veyjcqVW7AbIrIA)
